src/Microsoft.Health.Blob/Registration/BlobClientRegistrationExtensions.cs

```diff
@@ -29,43 +33,132 @@ public static IServiceCollection AddBlobDataStore(this IServiceCollection servic
                 return services;
             }
 
-            services.Add(provider =>
+            services.AddSingleton(
+                provider =>
                 {
                     var config = new BlobDataStoreConfiguration();
-                    provider.GetService<IConfiguration>().GetSection(BlobStoreConfigurationSectionName).Bind(config);
-                    configureAction?.Invoke(config);
+                    provider
+                        .GetService<IConfiguration>()
+                        .GetSection(BlobDataStoreConfiguration.SectionName)
+                        .Bind(config);
 
-                    if (string.IsNullOrEmpty(config.ConnectionString) && config.AuthenticationType == BlobDataStoreAuthenticationType.ConnectionString)
-                    {
-                        config.ConnectionString = BlobLocalEmulator.ConnectionString;
-                    }
+                    configureAction?.Invoke(config);
+                    DefaultBlobDataStoreConfiguration.Instance.Configure(config);
 
                     return config;
-                })
-                .Singleton()
-                .AsSelf();
+                });
 
-            services.Add<BlobClientProvider>()
-                .Singleton()
-                .AsSelf()
-                .AsService<IHostedService>() // so that it starts initializing ASAP
-                .AsService<IRequireInitializationOnFirstRequest>(); // so that web requests block on its initialization.
+            return configureAction == null
+                ? services.AddInitializedBlobServiceClient()
+                : services.AddInitializedBlobServiceClient(configureAction);
+        }
 
-            services.Add(sp => sp.GetService<BlobClientProvider>().CreateBlobClient())
-                .Singleton()
-                .AsSelf();
+        /// <summary>
+        /// Adds a singleton <see cref="BlobServiceClient"/> to the specified <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to be updated.</param>
+        /// <returns>The <paramref name="services"/> for additional method invocations.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="services"/> is <see langword="null"/>.</exception>
+        public static IServiceCollection AddBlobServiceClient(this IServiceCollection services)
+        {
+            EnsureArg.IsNotNull(services, nameof(services));
 
-            services.Add<BlobClientReadWriteTestProvider>()
-                .Singleton()
-                .AsService<IBlobClientTestProvider>();
+            services.AddOptions();
+            services.TryAddEnumerable(ServiceDescriptor.Singleton<IConfigureOptions<BlobDataStoreConfiguration>>(new DefaultBlobDataStoreConfiguration()));
```

Comment:
could we simplify this by using `ConfigureOptions`?
reference: https://andrewlock.net/simplifying-dependency-injection-for-iconfigureoptions-with-the-configureoptions-helper/